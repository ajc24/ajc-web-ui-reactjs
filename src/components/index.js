/**
 * Developed by Anthony Cox in 2024
 */

/* Export the header components */
export { default as HeaderTypeA } from './header/HeaderTypeA';

/* Export the main components */
export { default as MainContent } from './main/MainContent';

/* Export the menu bar components */
export { default as MenuBarTypeA } from './menu-bar/MenuBarTypeA';
