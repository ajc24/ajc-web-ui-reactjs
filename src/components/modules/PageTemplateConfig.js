/**
 * Developed by Anthony Cox in 2025
 */
import { storybookCssClassList } from '../data/storybook';

/* Commonly used CSS classes */
const bodyDefaultCss = 'body-default';

/**
 * Functionality designed to work with the page template components (headers, menu bars, body content, footers)
 */
export default class PageTemplateConfig {
  /**
   * Setup and configure the current web page to ensure the default CSS styling is set to it.
   * This also handles the styling of pages which are deployed on Storybook
   */
  static setupDocumentBodyCss() {
    /* If we are rendering on Storybook, remove the Storybook classes which will affect the width of the component */
    for (let index = 0; index < storybookCssClassList.length; index += 1) {
      if (document.body.classList.contains(storybookCssClassList[index]) === true) {
        document.body.classList.remove(storybookCssClassList[index]);
      }
    }
    /* Ensure that the <body> element stretches to 100% of the width of the screen, that the X axis scrollbar is hidden and Y axis scrollbar is always displayed */
    if (document.body.classList.contains(bodyDefaultCss) === false) {
      document.body.classList.add(bodyDefaultCss);
      document.activeElement.blur();
    }
  }
}
