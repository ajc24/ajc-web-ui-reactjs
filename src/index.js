/**
 * Developed by Anthony Cox in 2024
 */

/* Import the header components */
import Header from './components/header/Header';

/* Import the image components */
import DecorativeImage from './components/images/DecorativeImage';
import FunctionalImage from './components/images/FunctionalImage';
import InformativeImage from './components/images/InformativeImage';

/* Export all components */
export {
  DecorativeImage,
  FunctionalImage,
  InformativeImage,
  Header,
};
