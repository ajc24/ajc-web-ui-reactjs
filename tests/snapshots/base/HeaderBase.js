/**
 * Developed by Anthony Cox in 2025
 */
import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-testing-code';
import { HeaderBase } from '../../../src';
import PageTemplateConfig from '../../../src/components/modules/PageTemplateConfig';
import headerBgImage from '../../../stories/images/files/header-bg-image.png';
import tallHeaderBgImage from '../../../stories/images/files/tall-header-bg-image.png';

describe('Header Base', () => {
  /* Set the spies for use in the tests */
  let setupDocumentBodyCssSpy;

  beforeAll(() => {
    setupDocumentBodyCssSpy = jest
      .spyOn(PageTemplateConfig, 'setupDocumentBodyCss')
      .mockImplementation(() => {});
  });

  afterAll(() => {
    setupDocumentBodyCssSpy.mockRestore();
  });

  describe('Default component', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HeaderBase>
            Header base text content.
          </HeaderBase>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Component with custom id, small size, no background image, white background, gold top border', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HeaderBase id="custom-id" size="small" backgroundColour="white" topBorder="gold">
            Header base text content.
          </HeaderBase>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Component with default id, small size, no background image, grey background, green top border', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HeaderBase size="small" backgroundColour="grey" topBorder="green">
            Header base text content.
          </HeaderBase>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Component with default id, small size, with background image, default background, top border switched off', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HeaderBase size="small" backgroundImageSrc={headerBgImage} topBorder="off">
            Header base text content.
          </HeaderBase>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Component with default id, tall size, with background image, default background, green-2 top border', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HeaderBase size="tall" backgroundImageSrc={tallHeaderBgImage} topBorder="green-2">
            Header base text content.
          </HeaderBase>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Component with default id, tall size, no background image, default background, grey top border', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HeaderBase size="tall" topBorder="grey">
            Header base text content.
          </HeaderBase>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Component with default id, default size, no background image, default background, navy top border', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HeaderBase topBorder="navy">
            Header base text content.
          </HeaderBase>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Component with default id, default size, no background image, default background, red top border', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <HeaderBase topBorder="red">
            Header base text content.
          </HeaderBase>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
