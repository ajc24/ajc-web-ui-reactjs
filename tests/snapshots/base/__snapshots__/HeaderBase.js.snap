// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Base Component with custom id, small size, no background image, white background, gold top border verifies the snapshot for the component 1`] = `"<header role="banner" id="custom-id--header-base" class="header screen-width-root background-white"><div class="header-content-outer screen-width-content-outer background-white header-border-top-gold" data-bgimage="false"><div class="header-content-inner screen-width-content-inner header-small-height-with-border">Header base text content.</div></div></header>"`;

exports[`Header Base Component with default id, default size, no background image, default background, navy top border verifies the snapshot for the component 1`] = `"<header role="banner" id="default-id--header-base" class="header screen-width-root background-white"><div class="header-content-outer screen-width-content-outer background-white header-border-top-navy" data-bgimage="false"><div class="header-content-inner screen-width-content-inner header-small-height-with-border">Header base text content.</div></div></header>"`;

exports[`Header Base Component with default id, default size, no background image, default background, red top border verifies the snapshot for the component 1`] = `"<header role="banner" id="default-id--header-base" class="header screen-width-root background-white"><div class="header-content-outer screen-width-content-outer background-white header-border-top-red" data-bgimage="false"><div class="header-content-inner screen-width-content-inner header-small-height-with-border">Header base text content.</div></div></header>"`;

exports[`Header Base Component with default id, small size, no background image, grey background, green top border verifies the snapshot for the component 1`] = `"<header role="banner" id="default-id--header-base" class="header screen-width-root background-white"><div class="header-content-outer screen-width-content-outer background-grey header-border-top-green" data-bgimage="false"><div class="header-content-inner screen-width-content-inner header-small-height-with-border">Header base text content.</div></div></header>"`;

exports[`Header Base Component with default id, small size, with background image, default background, top border switched off verifies the snapshot for the component 1`] = `"<header role="banner" id="default-id--header-base" class="header screen-width-root background-white"><div class="header-content-outer screen-width-content-outer" data-bgimage="true"><div class="header-content-inner screen-width-content-inner header-small-height-no-border">Header base text content.</div></div></header>"`;

exports[`Header Base Component with default id, tall size, no background image, default background, grey top border verifies the snapshot for the component 1`] = `"<header role="banner" id="default-id--header-base" class="header screen-width-root background-white"><div class="header-content-outer screen-width-content-outer background-white header-border-top-grey" data-bgimage="false"><div class="header-content-inner screen-width-content-inner header-tall-height-with-border">Header base text content.</div></div></header>"`;

exports[`Header Base Component with default id, tall size, with background image, default background, green-2 top border verifies the snapshot for the component 1`] = `"<header role="banner" id="default-id--header-base" class="header screen-width-root background-white"><div class="header-content-outer screen-width-content-outer header-border-top-green-2" data-bgimage="true"><div class="header-content-inner screen-width-content-inner header-tall-height-with-border">Header base text content.</div></div></header>"`;

exports[`Header Base Default component verifies the snapshot for the component 1`] = `"<header role="banner" id="default-id--header-base" class="header screen-width-root background-white"><div class="header-content-outer screen-width-content-outer background-white" data-bgimage="false"><div class="header-content-inner screen-width-content-inner header-small-height-no-border">Header base text content.</div></div></header>"`;
